tools/git-version.gradle

```diff
@@ -0,0 +1,24 @@
+buildscript {
+    repositories {
+        jcenter()
+    }
+    dependencies {
+        classpath 'org.ajoberstar:grgit:1.9.2'
+    }
+}
+
+import org.ajoberstar.grgit.Grgit
+
+ext {
+    git = Grgit.open(currentDir: projectDir)
+    gitVersionName = ""
+    gitVersionName = git.describe()
```

Comment:
You can avoid adding another dependency by invoking git directly, and it's more similar to the actual git command. Like:
```
gitVersionName = 'git describe --abbrev=0 --tags HEAD^'.execute([], project.rootDir).text.trim()
```